homeassistant/components/powerwall/__init__.py

```diff
@@ -258,30 +261,33 @@ async def get_backup_reserve_percentage(power_wall: Powerwall) -> Optional[float
 
 async def _fetch_powerwall_data(power_wall: Powerwall) -> PowerwallData:
     """Process and update powerwall data."""
-    (
-        backup_reserve,
-        charge,
-        site_master,
-        meters,
-        grid_services_active,
-        grid_status,
-    ) = await asyncio.gather(
-        get_backup_reserve_percentage(power_wall),
-        power_wall.get_charge(),
-        power_wall.get_sitemaster(),
-        power_wall.get_meters(),
-        power_wall.is_grid_services_active(),
-        power_wall.get_grid_status(),
-    )
 
-    return PowerwallData(
-        charge=charge,
-        site_master=site_master,
-        meters=meters,
-        grid_services_active=grid_services_active,
-        grid_status=grid_status,
-        backup_reserve=backup_reserve,
-    )
+    try:
+        async with asyncio.TaskGroup() as tg:
+            backup_reserve = tg.create_task(get_backup_reserve_percentage(power_wall))
+            charge = tg.create_task(power_wall.get_charge())
+            site_master = tg.create_task(power_wall.get_sitemaster())
+            meters = tg.create_task(power_wall.get_meters())
+            grid_services_active = tg.create_task(power_wall.is_grid_services_active())
+            grid_status = tg.create_task(power_wall.get_grid_status())
+
+        return PowerwallData(
+            charge=charge.result(),
+            site_master=site_master.result(),
+            meters=meters.result(),
+            grid_services_active=grid_services_active.result(),
+            grid_status=grid_status.result(),
+            backup_reserve=backup_reserve.result(),
+        )
```

Comment:
I think we can move this block outside the `try` as we aren't concerned about `.result()` raising `BaseExceptionGroup` AFAICT